File: app/src/main/java/ltd/kaizo/moodtracker/model/MoodList.java
package ltd.kaizo.moodtracker.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

/**
 * The type Mood list.
 */
public class MoodList implements Serializable {
    /**
     * The Mood list.
     */
    private List<MoodItem> moodList;


    /**
     * Instantiates a new Mood list by initialize an arraylist of MoodItem
     */
    public MoodList() {
        this.moodList = new ArrayList();
    }

    /**
     * Add mood.
     *
     * @param mood the mood
     */
    public void addMood(MoodItem mood) {
        this.moodList.add(mood);
    }

    /**
     * Remove mood.
     *
     * @param mood the mood
     */
    public void removeMood(MoodItem mood) {
        this.moodList.remove(mood);
    }

    /**
     * Sets mood.
     *
     * @param index the index
     * @param mood  the mood
     */
    public void setMood(int index, MoodItem mood) {
        this.moodList.set(index, mood);
    }

    /**
     * Gets mood list.
     *
     * @return the mood list
     */
    public List<MoodItem> getMoodList() {
        return moodList;
    }

    /**
     * Gets mood item.
     *
     * @param index the index
     * @return the mood item
     */
    public MoodItem getMoodItem(int index) {
        return moodList.get(index);
    }

    /**
     * Gets size.
     *
     * @return the size
     */
    public int getSize() {
        return moodList.size();
    }

    /**
     * Compare to by date comparator.
     *
     * @return the comparator
     */
    public Comparator<MoodItem> compareToByDate() {
        Comparator comp = new Comparator<MoodItem>() {
            @Override
            public int compare(MoodItem o1, MoodItem o2) {
                return o1.getCurrentDate().compareTo(o2.getCurrentDate());
            }


        };
        return comp;
    }
}

File: app/src/main/java/ltd/kaizo/moodtracker/model/MoodList.kt
package ltd.kaizo.moodtracker.model

import java.io.Serializable
import java.util.*

/**
 * The type Mood list.
 */
class MoodList : Serializable {
    /**
     * The Mood list.
     */
    private val moodList: MutableList<MoodItem>

    val size: Int
        get() = moodList.size

    /**
     * Instantiates a new Mood list by initialize an arraylist of MoodItem
     */
    init {
        this.moodList = ArrayList()
    }

    /**
     * Add mood.
     *
     * @param mood the mood
     */
    fun addMood(mood: MoodItem) {
        this.moodList.add(mood)
    }

    /**
     * Remove mood.
     *
     * @param mood the mood
     */
    fun removeMood(mood: MoodItem) {
        this.moodList.remove(mood)
    }

    /**
     * Sets mood.
     *
     * @param index the index
     * @param mood  the mood
     */
    fun setMood(index: Int, mood: MoodItem) {
        this.moodList[index] = mood
    }

    /**
     * Gets mood list.
     *
     * @return the mood list
     */
    fun getMoodList(): List<MoodItem> {
        return moodList
    }

    /**
     * Gets mood item.
     *
     * @param index the index
     * @return the mood item
     */
    fun getMoodItem(index: Int): MoodItem {
        return moodList[index]
    }

    /**
     * Compare to by date comparator.
     *
     * @return the comparator
     */
    fun compareToByDate(): Comparator<MoodItem> {
        return Comparator { o1, o2 -> o1.currentDate!!.compareTo(o2.currentDate!!) }
    }
}
